examples/platform/esp32/lock/BoltLockManager.cpp

```diff
@@ -428,15 +428,7 @@ bool BoltLockManager::SetUser(chip::EndpointId endpointId, uint16_t userIndex, c
     userInStorage.lastModifiedBy = modifier;
     userInStorage.createdBy      = creator;
 
-    for (size_t i = 0; i < totalCredentials; ++i)
-    {
-        mCredentials[userIndex][i] = credentials[i];
-        // TODO: Why are we modifying the passed-in credentials?
-        // https://github.com/project-chip/connectedhomeip/issues/25083
-        // For now, preserve pre-existing behavior, which set credentialType to 1.
-        mCredentials[userIndex][i].credentialType  = CredentialTypeEnum::kPin;
-        mCredentials[userIndex][i].credentialIndex = i + 1;
-    }
+    memcpy(mCredentials, credentials, totalCredentials * sizeof(CredentialStruct));
```

Comment:
Can you explain to me, in detail, why this memcpy is the right thing to do?

Reply:
memcpy is fine for copying credentials _if you are copying to the right memory_. Which the above code was not.